### PR TITLE
OBS-158: Add github actions CI

### PR DESCRIPTION
**Pull request by relud**
add github actions matching circleci flow to push to antenna image to GAR for gcp deploy flow. CircleCI will remain until aws is gone.

### CODE REVIEW

File: .github/workflows/build-and-push.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          # https://github.com/mozilla-services/Dockerflow/blob/master/docs/version_object.md
+          printf '{"commit":"%s","version":"%s","source":"%s","build":"%s"}\n' \
+          "$GITHUB_SHA" \
+          "$GITHUB_REF_NAME" \
+          "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY" \
+          "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID" > version.json
+      - name: Output version.json
+        run: cat version.json
+      - name: Build Docker images
+        run: make build
```
Comment (relud) [inline .github/workflows/build-and-push.yml:39]: in antenna's circleci we log in to docker before this step, but in eliot we don't.

I'm not sure which pattern to follow here.

**Review by relud — COMMENTED** [review]
`.github/workflows/build-and-push.yml` was copied from eliot and then steps were modified to match antenna's circleci config.

ceverything that is modified from eliot is explicitly called out in comments.

File: .github/workflows/build-and-push.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          echo DEPLOYMENT_ENV=prod >> "$GITHUB_ENV"
+      - name: Push the Docker image to GAR
+        if: env.IMAGE_TAG != ''
+        uses: mozilla-it/deploy-actions/docker-push@v3.9.0
+        with:
+          local_image: local/antenna_deploy_base:latest
+          image_repo_path: ${{ secrets.DOCKER_IMAGE_PATH }}
+          image_tag: ${{ env.IMAGE_TAG }}
+          workload_identity_pool_project_number: ${{ secrets.WORKLOAD_IDENTITY_POOL_PROJECT_NUMBER }}
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
```
Comment (relud) [inline .github/workflows/build-and-push.yml:92]: I configured these secrets, but i don't know if i got them right, i will need to confirm with @smarnach when he is back

Comment (relud) [reply .github/workflows/build-and-push.yml:92]: figured it out

File: .github/workflows/build-and-push.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
+name: Build, test and push a Docker image
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v20[0-9][0-9].[01][0-9].[0-3][0-9]  # e.g. v2023.12.04
+      - v20[0-9][0-9].[01][0-9].[0-3][0-9]-[0-9]  # e.g. v2023.12.04-2
```
Comment (willkg) [inline .github/workflows/build-and-push.yml:9]: Currently Antenna uses a `YYYY.MM.DD[-N]` tag scheme. We haven't changed that, yet. 

* [bug 1882879: changing Antenna tag scheme](https://bugzilla.mozilla.org/show_bug.cgi?id=1882879)
* [bug 1857938: changing Socorro tag scheme](https://bugzilla.mozilla.org/show_bug.cgi?id=1857938)

We should probably do those before setting up gha based deploys.

File: .github/workflows/build-and-push.yml (hunk 1 above)
Comment (willkg) [reply .github/workflows/build-and-push.yml:39]: Antenna/AWS/CircleCI/DockerHub pushes images to Docker Hub.

For Antenna/GCP/GHA/GAR, we don't need to push images to Docker Hub, so I don't think we need to log into Docker Hub. That was the reasoning for not logging into Docker Hub with Eliot.

**Review by willkg — APPROVED** [review]
Except for the fact that this is changing the tag schema and we should tackle that bug first, this looks good to me.

**Comment by willkg** [comment]
PR 1000! :tada:

File: .github/workflows/build-and-push.yml (hunk 1 above)
Comment (relud) [reply .github/workflows/build-and-push.yml:39]: cool, we'll continue with that then, and add docker login for rate limiting only if it becomes an issue

**Pull request merged**